### PR TITLE
Locks use a different command

### PR DESCRIPTION
**Pull request by Teagan42**
For now, I'm sort of hacking your library to get locks to work in HASS..

``` python
    def lock(self, **kwargs) -> None:
        """Lock the device."""
        # Hack until PyISY is updated
        req_url = self._conn.compileURL(['nodes', self.unique_id, 'cmd',
                                         'SECMD', '1'])
        response = self._conn.request(req_url)

        if response is None:
            _LOGGER.error('Unable to lock device')

        self._node.update(0.5)

    def unlock(self, **kwargs) -> None:
        """Unlock the device."""
        # Hack until PyISY is updated
        req_url = self._conn.compileURL(['nodes', self.unique_id, 'cmd',
                                         'SECMD', '0'])
        response = self._conn.request(req_url)

        if response is None:
            _LOGGER.error('Unable to lock device')

        self._node.update(0.5)
```

Also added support for aux. properties (battery level, humidity, set points, etc.) and now the commands for climate (thermostats)!

### CODE REVIEW

**Comment by Teagan42** [comment]
I didn't like only having 3 units of measure in there - figure leave it up to the consumer to parse out the uom code via the docs.

**Comment by Teagan42** [comment]
Looks like some devices have multiple properties - like the Z-Wave Motion sensor has a battery level and an on/off - guess yet another PR for that

**Comment by Teagan42** [comment]
Don't merge yet - gonna change how this does the ancillary properties

**Comment by jimboca** [comment]
@Teagan42 if you can resolve these conflicts with my property changes that @rmkraus had approved, then I will test your changes in my environment to see if it fixes an issue I had.

**Comment by Teagan42** [comment]
So - I was hoping to try to get all of the sub-properties from nodes. Some nodes have their state and then have a battery level or what have you. That's why I was trying to allow for auxillary properties. I haven't looked at this code in a few months, so let me know if there's anything you don't understand.

**Comment by jimboca** [comment]
The changes all look good to me, but since they are so involved I will have to defer to @rmkraus to merge this.

**Comment by hoopty** [comment]
What else needs to be done for this to be merged?

**Comment by Teagan42** [comment]
@hoopty it needs reviewed/approved by repo members

File: PyISY/Nodes/node.py [hunk 1]
```diff
@@ -54,11 +54,23 @@ def parse_xml_properties(xmldoc):
     if len(props) > 0:
         for prop in props:
             attrs = prop.attributes
-            prop_id = attrs[ATTR_ID].value if ATTR_ID in attrs else None
```
Comment (jimboca) [inline PyISY/Nodes/node.py]: What about using something like:
if ATTR_ID in prop.attributes.keys():

**Comment by jimboca** [comment]
@Teagan42 @rmkraus asked me to take a look at this since he can not right now.  Everything looks okay to me except my one comment about all the exception based coding, so try to figure out a better way to handle that and I'll merge when I can.

**Comment by Teagan42** [comment]
Your suggestion worked - I'm just testing a couple more things.

**Comment by jimboca** [comment]
great, I'm running in my environment and will try and merge if all goes well.

**Comment by Teagan42** [comment]
FYI - the aux properties only seem to come in if you run
`ISY.nodes.update()` - I haven't quite figured out a good way that is
backwards compatible to have the aux properties come over with the auto
updates

**Comment by jimboca** [comment]
How will this not be backwards compatible?  The only way to get the properties is a full refresh since they do not change on the fly, right?

**Comment by Teagan42** [comment]
The current code in this pull request is backwards compatible however I was
trying to find a way to automatically update the auxiliary properties
similar to how the state is updated automatically but the ways I found
would break backwards compatibility.

**Comment by Teagan42** [comment]
@jimboca When do you think PyPI will be updated?

**Comment by Teagan42** [comment]
I actually have some more changes I'd like to add - so if you wanna wait on PyPI that'd be awesome :-)

**Comment by jimboca** [comment]
Sorry, @rmkraus will have release.

**Pull request merged**